pymysqlreplication/row_event.py

```diff
@@ -55,7 +55,7 @@ def __init__(self, from_packet, event_size, table_map, ctl_connection, **kwargs)
         self.columns = self.table_map[self.table_id].columns
 
         if len(self.columns) == 0:  # could not read the table metadata, probably already dropped
-            self.complete = False
+            raise TableMetadataUnavailableError("Table Metadata could not be read for {}".format(self.table))
```

Comment:
Maybe this should be optional?  I absolutely want it, but the maintainer didn't - I think it's much more correct to raise an exception, but if we want to contribute back upstream, we probably shouldn't force the issue.